DM-27508 Allow collection name to be specified in DBB BM Gen3 Ingester config

mxk62 opened the pull request.

mxk62 committed:
Increase number of pass through options

Only a transfer mode could be passed through to both Gen2 and Gen3 LSST
ingest software.  I added support for passing through additional
settings.

mxk62 committed:
Add version related attributes to the package

mxk62 committed:
Fix logging

Some log messages from LSST Gen3 ingest software were displayed on
stdout/stderr despite setting up Python logger to use a file.  Made a
change to address this issue.

mxk62 committed:
Allow for using tables in different schemas

The object relational mappers (ORMs) used by the endpoint manager
couldn't be associated with tables located "outside" of the default
database schema.  I extended ORMs definitions to allow for specifying
not only the table name, but the schema as well.

mxk62 committed:
Update example configuration files

Updated example configuration files to reflect the new features.

mxk62 committed:
Add example configuration for Gen3 ingester

Only the example configuration for Gen2 Ingester was provided.  Added
one for Gen3 Ingester.

mxk62 committed:
Update documentation

Updated the documentation of the package to make it reflect the recent
changes.

MichelleGower commented on doc/lsst.dbb.buffmngrs.endpoint/get-started.rst:
```diff
@@ -79,8 +79,8 @@ component that you may adapt to your local configuration:
      tablenames:
-       file: files
-       event: gen2_file_events
+       file:
```
Documentation for schema key in the etc/*.yaml files?

MichelleGower commented on python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py:
```diff
@@ -127,62 +144,69 @@ def execute(self, filename):
+    The LSST Gen3 Middleware presumably supports parallel ingestion of
+    multiple files.  However, it does not expose information which files
+    were ingested successfully and which failed to ingest.  Hence,
+    the plugin is deliberately restricted to ingesting one file at the time.
```
Parallelism is instead done at the ? level by setting ?.

mxk62 replied:
I'll add

    To enable concurrent ingests, use Ingester's ``num_threads`` setting  instead.

to hint how to enable concurrent ingests in the manager.

pehartma commented on python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py:
```diff
@@ -127,62 +144,69 @@ def execute(self, filename):
+        if self.config["config"] is not None:
+            for key, val in config.items():
+                config_overrides.addValueOverride(key, val)
+        self.task = class_(config=config, butler=butler)
```
Needs "config_overrides.applyTo(config)"

mxk62 replied:
Indeed! Thanks for catching it.

MichelleGower commented on python/lsst/dbb/buffmngrs/endpoint/ingester/commands.py:
```diff
@@ -86,37 +86,48 @@ def start(filename, dump, validate):
-        missing = required - available
-        if missing:
-            msg = f"table(s) {', '.join(missing)} not found in the database"
+    required = set()
```
Are these new lines (~92-117) the same as in the finder/commands.py?  If so, think about putting them in a separate function.

mxk62 replied:
As suggested, I rewrote the repeated code as functions and put them in `utils.py` so both the Finder and the Ingester can use them.

MichelleGower commented on python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py:
```diff
@@ -20,10 +20,12 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 import importlib
```
importlib no longer used?

MichelleGower reviewed (approved):
Some minor comments.  After changes, good to merge.

mxk62 committed:
Factor out checking table existence

The code responsible for checking if the required table exist in the
database was repeated in both the Finder and the Ingester.  I factored it
out to a function and put it in the utils.py module.
